src/zen/websocket_server.cpp

```diff
@@ -488,6 +508,65 @@ class WsHandler
         return OK;
     }
 
+    int sendTopQualityCertificatesForScid(const std::string& scIdString, const std::string& clientRequestId)
+    {
+        uint256 scId;
+        scId.SetHex(scIdString);
+        
+        const CCoinsViewCache &view = *pcoinsTip;
+        
+        UniValue mempoolTopQualityCert(UniValue::VOBJ);
+        UniValue chainTopQualityCert(UniValue::VOBJ);
+
+        {
+            LOCK(cs_main);
+            if (!view.HaveSidechain(scId)) {
+                LogPrint("ws", "%s():%d - sidechain id not found[%s]\n", __func__, __LINE__, scIdString);
+                return INVALID_PARAMETER;
+            }
+        
+            if (mempool.hasSidechainCertificate(scId))
+            {
+                const uint256& topQualCertHash = mempool.mapSidechains.at(scId).GetTopQualityCert()->second;
+                const CScCertificate& topQualCert = mempool.mapCertificate.at(topQualCertHash).GetCertificate();
+                const CAmount certFee = mempool.mapCertificate.at(topQualCertHash).GetFee();
+                CDataStream ssCert(SER_NETWORK, PROTOCOL_VERSION);
+                ssCert << topQualCert;
+                std::string certHex = HexStr(ssCert.begin(), ssCert.end());
+     
+                mempoolTopQualityCert.push_back(Pair("quality", topQualCert.quality));
+                mempoolTopQualityCert.push_back(Pair("certHash", topQualCertHash.GetHex()));
+                mempoolTopQualityCert.push_back(Pair("rawCertificateHex", certHex));
+                mempoolTopQualityCert.push_back(Pair("fee", FormatMoney(certFee)));
+            }
+
+            CSidechain sidechainInfo;
+
+            if (view.GetSidechain(scId, sidechainInfo) && !sidechainInfo.lastTopQualityCertHash.IsNull()) {
+                const int topQualityCertQuality = sidechainInfo.lastTopQualityCertQuality;
+                CScCertificate topQualCert;
+                uint256 blockHash;
+
+                if (GetCertificate(sidechainInfo.lastTopQualityCertHash, topQualCert, blockHash)) {
+                    CDataStream ssCert(SER_NETWORK, PROTOCOL_VERSION);
+                    ssCert << topQualCert;
+                    std::string certHex = HexStr(ssCert.begin(), ssCert.end());
+
+                    chainTopQualityCert.push_back(Pair("quality", topQualityCertQuality));
```

Comment:
return also as a separate field the certificate referenced epoch (we have to be sure on sidechain side that the last received certificate doesn't belong to the previous epoch)
Do the same for mempool cert